### PR TITLE
Testnet integration

### PR DESCRIPTION
**Pull request by foultre**
Integration with Corda Testnet network

### CODE REVIEW

File: TESTNET.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# How to set up a corda k8s node on the corda testnet network
```
Comment (henrikr3) [inline TESTNET.md]: Please merge this file with CHECKLIST.md.

File: TESTNET.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Azure steps :
```
Comment (henrikr3) [inline TESTNET.md]: CHECKLIST.md already has cloud infrastructure setup steps, this part is not needed.

File: TESTNET.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - node-storage (2Gb)
+  - bridge-storage (1Gb)
+  - float-storage (1Gb)
+- create an ACR (docker registry)
+- create 2 public IP adresses in the same RG as AKS (starts with MC_)
+  - p2p-ip
+  - rpc-ip
+- fill `helm/values.yaml` with appropriate values from above
+
+Repo steps :
```
Comment (henrikr3) [inline TESTNET.md]: This should be integrated to CHECKLIST.md, with the following comments.

File: TESTNET.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+- Do docker steps (docker directory)
```
Comment (henrikr3) [inline TESTNET.md]: Change values.yaml initialRegistration to false, then execute one-time-setup.sh.

File: TESTNET.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- Register on corda marketplace : https://marketplace.r3.com/register
```
Comment (henrikr3) [inline TESTNET.md]: The following block should be integrated to the CHECKLIST.md, but with a heading for Testnet configuration

File: TESTNET.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  - Corda version : 4.0
+- Click on "Create new node"
+- Click on download corda node. It should download a `node.zip` file
+- unzip the `node.zip` on a PC or a VM with internet access. (We are going to call it NODE_DIR)
+- Edit `node.conf` and change `p2pAddress` 
+- Run in a shell: `java -jar corda.jar` 
+- Wait for node to start. That will enrich the certificate with the node legal entity key pair
+- kill the node process
+- Copy certificates `cp $NODE_DIR/certificates/*.jks ./helm/files/certificates/node`
+- Fill variables  `keystorePassword` and `truststorePassword` from unzipped `node.conf` to `helm/values.yaml` `nodeKeystorePassword` and `nodeTruststorePassword`
```
Comment (henrikr3) [inline TESTNET.md]: Update variables based on other comments in this PR

File: TESTNET.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+- Fill variable `myLegalName` from node.conf to `helm/values.yaml` `corda.node.conf.legalName`
+- Fill variable `corda.node.conf.compatibilityZoneEnabled`  with `true` in `helm/values.yaml` 
+- Fill variable `corda.node.conf.compatibilityZoneURL` with `https://netmap.testnet.r3.com` in `helm/values.yaml` 
+- `kubectl create namespace cordatest`
```
Comment (henrikr3) [inline TESTNET.md]: This part is not related to testnet and already exists in CHECKLIST.md

File: helm/files/bridge.conf [hunk 8]
```diff
@@ -30,8 +30,8 @@ bridgeInnerConfig: {
 certificatesDirectory: "./workspace/certificates"
 sslKeystore: "./workspace/certificates/sslkeystore.jks"
 trustStoreFile: "./workspace/certificates/truststore.jks"
-keyStorePassword: "cordacadevpass"
-trustStorePassword: "{{ .Values.corda.firewall.conf.truststorePassword }}"
+keyStorePassword: "{{ .Values.corda.nodebridge.conf.nodeKeystorePassword }}"
+trustStorePassword: "{{ .Values.corda.nodebridge.conf.nodeTruststorePassword }}"
```
Comment (henrikr3) [inline helm/files/bridge.conf]: No need to introduce a new nodebridge group, we already have firewall group which defines all conf changes to bridge/float conf files.

File: helm/files/node.conf [hunk 9]
```diff
@@ -1,10 +1,14 @@
 myLegalName: "{{ .Values.corda.node.conf.legalName }}"
 emailAddress: "{{ .Values.corda.node.conf.emailAddress }}"
 p2pAddress: "{{ .Values.corda.node.conf.p2pAddress }}:{{ .Values.corda.node.conf.p2pPort }}"
+{{- if .Values.corda.node.conf.compatibilityZoneEnabled }}
+compatibilityZoneURL: ".Values.corda.node.conf.compatibilityZoneURL"
+{{ else }}
```
Comment (henrikr3) [inline helm/files/node.conf]: {{- else }}

File: helm/files/node.conf [hunk 10]
```diff
@@ -1,10 +1,14 @@
...
 networkServices: {
   doormanURL: "http://{{ .Values.corda.node.conf.identityManagerAddress }}",
   networkMapURL: "http://{{ .Values.corda.node.conf.networkmapAddress }}"
 }
+{{ end }}
```
Comment (henrikr3) [inline helm/files/node.conf]: {{- end }}

File: helm/values.yaml [hunk 11]
```diff
@@ -125,6 +125,10 @@ corda:
       identityManagerAddress: ""
       # networkmapAddress defines the accesspoint for the Network Map server (public domain name+port)
       networkmapAddress: ""
+      # compatibilityZoneEnabled set to true for testnet
```
Comment (henrikr3) [inline helm/values.yaml]: compatibilityZoneEnabled defines if you are connecting to a compatibility zone rather than using the above identityManagerAddress+networkmapAddress. Set this to enabled to be able to join Corda Testnet along with compatibilityZoneURL.

File: helm/values.yaml [hunk 12]
```diff
@@ -125,6 +125,10 @@ corda:
...
+      compatibilityZoneEnabled: false
+      # compatibilityZoneURL The root address of Corda compatibility zone network management services
```
Comment (henrikr3) [inline helm/values.yaml]: compatibilityZoneURL defines the root address of Corda compatibility zone network management services. For Corda Testnet you can use "https://netmap.testnet.r3.com"

File: helm/values.yaml [hunk 13]
```diff
@@ -175,6 +179,12 @@ corda:
       allowDevCorDapps:
         # enabled is a boolean value for the above parameter
         enabled: true
+  # nodebridge is the options that define how to define the node.conf and bridge.conf files which defines the bridge/node tunnel
```
Comment (henrikr3) [inline helm/values.yaml]: This section is extra, we already have firewall section below, which should be used instead.

File: TESTNET.md (hunk 1 above)
Comment (foultre) [reply TESTNET.md]: done

File: TESTNET.md (hunk 2 above)
Comment (foultre) [reply TESTNET.md]: done
